gym/vector/async_vector_env.py

```diff
@@ -717,7 +737,10 @@ def _worker_shared_memory(index, env_fn, pipe, parent_pipe, shared_memory, error
                     pipe.send((None, True))
             elif command == "step":
                 observation, reward, done, info = env.step(data)
-                if done:
+                if isinstance(env, VectorEnv):
+                    # VectorEnvs take care of resetting the envs that are done.
+                    pass
+                elif done:
```

Comment:
Would it maybe be better to just do a single condition `if done and isinstance(env, Env)` or `if done and not isinstance(env, VectorEnv)`? (not sure which would be clearer, but the `pass` gave me a bit of a pause since it's rarely present in released code, and it felt more like something unfinished)

Reply:
Fixed in 055087b6ae7711f9565f681b2f6befe563f6ba3e